### PR TITLE
Add local (per-project) swiftlint path setting.

### PR DESCRIPTION
**Pull request by MontakOleg**
Implemented https://github.com/bealex/SwiftLintAppCode/issues/38 by adding new local(per-project) swiftlint path setting. It will be used if specified, otherwise global swiftlint path will be used.

### CODE REVIEW

File: src/com/lonelybytes/swiftlint/Configuration.java [hunk 1]
```diff
@@ -69,10 +72,31 @@ public JComponent createComponent() {
         row.add(browser);
         panel.add(row);
 
+        // Local path
+
+        JTextField localPathTextField = new JTextField(30);
+        JLabel localPathLabel = new JLabel("Per-project SwiftLint path:");
```
Comment (MontakOleg) [inline src/com/lonelybytes/swiftlint/Configuration.java:78]: Maybe we should add "optional" in title?

**Comment by bealex** [comment]
@MontakOleg thanks for the PR! It is very helpful.

One addition that I'd like to see. Setting itself must be marked as "For current project" (to be consistent with other preferences). Or whole settings pane should be per-project. I'll look into it myself, but maybe you know how to do that?

**Comment by MontakOleg** [comment]
@bealex thanks for quick feedback!

Unfortunately I don't know. Maybe `projectConfigurable` Extension Point can help.

**Comment by bealex** [comment]
I've asked JB guys about it, here is the answer. Saving it here just to be able to refer to it. I'll update code accordingly.

> Если вкратце, то нужно унаследоваться от Configurable или SearchableConfigurable и задекларировать в xml через extension point projectConfigurable. Вот здесь документация на эту тему: https://www.jetbrains.org/intellij/sdk/docs/reference_guide/settings_guide.html#declaring-project-settings
> 
> Вот тут я нашёл случайный опенсорсный плагин с настройками, можно посмотреть, как именно это сделано: https://github.com/JetBrains/intellij-plugins/blob/7255fb3144b49aa86483ffad42cf79d8b6452e59/tslint/resources/META-INF/plugin.xml#L27

**Comment by MontakOleg** [comment]
I think it can be confused if we left two settings (global SwiftLint path & per-project path) AND mark whole settings pane as "for current project". I made changes locally, see how it look:
settings

Maybe it would be better to make all settings project level (with reasonable default for path like `/usr/local/bin/swiftlint`) and left only one "SwiftLint path".

It should be as simple as changing `PropertiesComponent.getInstance()` to `PropertiesComponent.getInstance(getProject())` if we don't care about settings migration from old versions.

**Comment by bealex** [comment]
We do care about migration, but yes, this is exactly what I was aiming for.

**Comment by bealex** [comment]
Added this functionality to 1.13

**Pull request closed without merging**